Arrowgene.Ddon.GameServer/Handler/ServerGameTimeGetBaseinfoHandler.cs

```diff
@@ -1,30 +1,34 @@
-﻿using System;
 using Arrowgene.Ddon.Server;
-using Arrowgene.Ddon.Server.Network;
 using Arrowgene.Ddon.Shared.Entity.PacketStructure;
-using Arrowgene.Ddon.Shared.Network;
+using Arrowgene.Ddon.Shared.Entity.Structure;
 using Arrowgene.Logging;
 
 namespace Arrowgene.Ddon.GameServer.Handler
 {
-    public class ServerGameTimeGetBaseinfoHandler : StructurePacketHandler<GameClient, C2SServerGameTimeGetBaseInfoReq>
+    public class ServerGameTimeGetBaseinfoHandler : GameRequestPacketHandler<C2SServerGameTimeGetBaseInfoReq, S2CServerGameTimeGetBaseInfoRes>
     {
         private static readonly ServerLogger Logger = LogProvider.Logger<ServerLogger>(typeof(ServerGameTimeGetBaseinfoHandler));
 
         public ServerGameTimeGetBaseinfoHandler(DdonGameServer server) : base(server)
         {
         }
 
-        public override void Handle(GameClient client, StructurePacket<C2SServerGameTimeGetBaseInfoReq> packet)
+        public override S2CServerGameTimeGetBaseInfoRes Handle(GameClient client, C2SServerGameTimeGetBaseInfoReq request)
         {
-            client.Send(new S2CServerGameTimeGetBaseInfoRes());
-        }
+            var res = new S2CServerGameTimeGetBaseInfoRes();
+            res.GameTimeBaseInfo.GameTimeOneDayMin = Server.Setting.GameLogicSetting.GameClockTimescale;
+            res.WeatherLoop = Server.WeatherManager.WeatherLoopList;
 
-        // Adapted from the client's code
-        private long calcGameTimeMSec(DateTimeOffset realTime, long originalRealTimeSec, uint gameTimeOneDayMin, uint gameTimeDayHour)
-        {
-            return (1440 * (realTime.Millisecond + 1000 * (realTime.ToUnixTimeSeconds() - originalRealTimeSec)) / gameTimeOneDayMin)
-            % (3600000 * gameTimeDayHour);
+            //TODO: Investigate these values. The moon cycles but predicting the current phase serverside is still unclear.
+            res.MoonAgeLoopSec = Server.Setting.GameLogicSetting.GameClockTimescale * 60;
+            res.MoonSchedule.Add(new CDataMoonSchedule()
+            {
+                BeginTimeSec = long.MinValue,
+                EndTimeSec = long.MaxValue,
+                MoonAge = 14
```

Comment:
This moon age is interesting. If it is supposed to mean phases, I thought there is generally 8 accepted phases of the moon. Or does age mean days and maybe associated with time in a lunar calendar? Did you find any pattern for this?